src/api/layers/VectorLayer.class.js

```diff
@@ -0,0 +1,27 @@
+import GeoAdminLayer from '@/api/layers/GeoAdminLayer.class'
+import LayerTypes from '@/api/layers/LayerTypes.enum'
+
+/** Metadata for a vector tile layer (MapLibre layer) */
+export default class VectorLayer extends GeoAdminLayer {
+    /**
+     * @param {string} layerId The ID of this layer
+     * @param {number} opacity The opacity of this layer, between 0.0 (transparent) and 1.0 (opaque)
+     * @param {string} styleUrl The URL to access the style (Mapbox style JSON)
+     */
+    constructor(layerId, opacity, styleUrl) {
+        super(
+            layerId,
+            LayerTypes.VECTOR,
+            layerId,
+            opacity,
+            'swisstopo',
+            'https://www.swisstopo.admin.ch/en/home.html',
+            true
```

Comment:
Are all these options of the constructor hardcoded because the backend does not provide them? If yes should the class name maybe be a bit more specific (like swisstopoBackgroundVectorLayer) or these options defined when the constructor is called?

Reply:
That's a good point that was also raised by Brice, it should now be more generic, with hardcoded value only found in one place